cmd/log-shuttle/main.go

```diff
@@ -244,8 +244,9 @@ func main() {
 
 	go LogFmtMetricsEmitter(s.MetricsRegistry, config.StatsSource, config.StatsInterval, s.Logger)
 
-	// Blocks until os.Stdin is closed
+	// Blocks until os.Stdin errors
 	s.ReadLogLines(os.Stdin)
```

Comment:
I think it's the case that I don't understand the intended purpose.... 

What happens when you add a Reader using LoadReader() and *it* errors out?